tests/test_error_checking.py

```diff
@@ -25,3 +26,11 @@ def test_stability(expr):
 
     with pytest.raises(ExecutionError):
         op.apply(time_M=200, dt=.1)
+
+@pytest.mark.parametrize("cc", [
+    "doesn'texist",
+    "/root/doesn'texist",
+])
+def test_sniff_compiler_version(cc):
+    with pytest.raises(SystemExit) as e:
+        sniff_compiler_version(cc)
```

Comment:
I think it needs  to be

```
with pytest.raises(SystemExit) as e:
        sniff_compiler_version(cc)
 assert e == SystemExit
```

so there is a test.

Reply:
Also: this test should be moved to some other files as this one, `test_error_checking`, is supposed to be for runtime (C-level) numerical and computational errors.

Perhaps `test_environment`?

Reply:
`sys.exit(...)` is also quite brutal, perhaps we turn it into a RuntimeError?

Reply:
`test_environment.py` seems to be making sure the test environment is usable, so this test doesn't fit there in my opinion. This test is independent of the test environment. Should there be a second test file for non-numerical runtime errors?

I agree that RuntimeError is better. I'll make that change.

Reply:
> Should there be a second test file for non-numerical runtime errors?

perhaps `test_arch.py` where in the future we can also add all other `/devito/arch` related tests?

Reply:
I moved the test to `test_arch.py` and fixed the pep8 errors, so as far as I know, this is ready to be merged.

Reply:
There is benchmark setup that needs fixing and rebase, I'll do it and get it merged